src/harness/fourslash.ts

```diff
@@ -627,26 +632,60 @@ namespace FourSlash {
             }
         }
 
-        private verifyGoToXSingle(startMarkerName: string, endMarkerNames: string | string[], getDefs: () => ts.DefinitionInfo[] | undefined) {
+        private verifyGoToXSingle(startMarkerName: string, endMarkerNames: string | string[], getDefs: () => ts.DefinitionInfo[] | ts.DefinitionInfoAndBoundSpan | undefined) {
             this.goToMarker(startMarkerName);
-            this.verifyGoToXWorker(toArray(endMarkerNames), getDefs);
+            this.verifyGoToXWorker(toArray(endMarkerNames), getDefs, startMarkerName);
         }
 
-        private verifyGoToXWorker(endMarkers: string[], getDefs: () => ts.DefinitionInfo[] | undefined) {
-            const definitions = getDefs() || [];
+        private verifyGoToXWorker(endMarkers: string[], getDefs: () => ts.DefinitionInfo[] | ts.DefinitionInfoAndBoundSpan | undefined, startMarkerName?: string) {
+            const defs = getDefs();
+            let definitions: ts.DefinitionInfo[] | ReadonlyArray<ts.DefinitionInfo>;
+            let testName: string;
+            if (this.isDefinitionInfoAndBoundSpan(defs)) {

```

Comment:
Would it be easier to just check `isArray`?

Reply:
It would be the same IMO, if anything maybe a little bit harder to read.

First, we need to compare against undefined as well making the comparison a little bit bigger. Second, we'll need to use a type assertion instead of a type guard.

Reply:
`isArray` has the advantage of already existing.  And I believe (but don't know) that code flow type inference will make type assertions unnecessary (essentially, any[] & (foo[] | bar) => foo[]).

Reply:
Yes, ```isArray``` will narrow the type, but due to the ```undefined``` check it will not be able to do it. In any case, I have added the modification as it's no big deal on readability and like you mentioned already exists and no custom guards are necessary.